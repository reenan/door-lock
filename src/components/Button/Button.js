import React, { Component } from 'react'
import PropTypes from 'prop-types'

import { Icon } from '../'
import './Button.scss'

class Button extends Component {
  render() {
    const { className, style, onClick, icon, text, size } = this.props

    if (size) {
      style.width = size + 'px'
      style.height = size + 'px'
      style.lineHeight = size + 'px'
    }

    return (
      <div className={`button ${className}`} style={style} onClick={onClick}>
        {
          text ? <span>{text}</span> : null
        }
        {
          icon ? <Icon size={size} icon={icon} /> : null
        }
      </div>
    )
  }
}

Button.propTypes = {
  className: PropTypes.string,
  size: PropTypes.number,
  onClick: PropTypes.func,
  style: PropTypes.object,
  icon: PropTypes.string,
  text: PropTypes.string,
}

Button.defaultProps = {
  className: '',
  size: null,
  icon: null,
  text: null,
  style: {},
}

export default Button
