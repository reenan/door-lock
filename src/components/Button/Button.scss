@import '../../resources/_mixins.scss';

.button {
  @include vertical-align(middle);
  @include transition(background-color 400ms, color 400ms);
  cursor: pointer;
  color: #25cb68;
  font-size: 14px;
  text-transform: uppercase;
  padding: 10px 15px;

  span {
    @include vertical-align(middle);

    + span {
      margin-left: 15px;
    }
  }

  .icon {
    font-size: 22px;
  }

  &:hover {
    color: darken(#25cb68, 10);
  }

  &.disabled {
    color: gray;
    pointer-events: none;
  }
}