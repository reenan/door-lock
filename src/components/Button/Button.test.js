import React from 'react'
import Button from './Button'

import renderer from 'react-test-renderer'

it('renders without crashing', () => {
  const component = renderer.create(
    <Button />
  )

  expect(component).toMatchSnapshot()
})

it('renders without crashing with size', () => {
  const component = renderer.create(
    <Button size={11} />
  )

  expect(component).toMatchSnapshot()
})

it('renders without crashing with text and icon', () => {
  const component = renderer.create(
    <Button text='Mock' icon='Mock' />
  )

  expect(component).toMatchSnapshot()
})

