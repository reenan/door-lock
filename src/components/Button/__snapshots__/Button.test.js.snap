// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="button "
  style={Object {}}
/>
`;

exports[`renders without crashing with size 1`] = `
<div
  className="button "
  style={
    Object {
      "height": "11px",
      "lineHeight": "11px",
      "width": "11px",
    }
  }
/>
`;

exports[`renders without crashing with text and icon 1`] = `
<div
  className="button "
  style={
    Object {
      "height": "11px",
      "lineHeight": "11px",
      "width": "11px",
    }
  }
>
  <span>
    Mock
  </span>
  <span
    className="icon Mock "
    style={Object {}}
  />
</div>
`;
